aodncore/pipeline/handlerbase.py

```diff
@@ -1022,15 +1022,21 @@ def postprocess(self):  # pragma: no cover
     # "public" methods
     #
 
-    def add_pipelinefile(self, pipeline_file):
-        """Helper method to add a PipelineFile to the handler's file_collection, with the handler instance's
-            file_update_callback method assigned
+    def add_to_collection(self, pipeline_file, **kwargs):
+        """Helper method to add a PipelineFile object or path to the handler's file_collection, with the handler
+        instance's file_update_callback method assigned
 
-        :param pipeline_file: PipelineFile object
+            Note: as this is a wrapper to the PipelineFileCollection.add method, kwargs are *only* applied when
+            pipeline_file is a path (string). When adding an existing PipelineFile object, the object is added "as-is",
+            and attributes must be set explicitly on the object.
+
+        :param pipeline_file: :py:class:`PipelineFile` or file path
+        :param kwargs: keyword arguments passed through to the PipelineFileCollection.add method
         :return: None
         """
-        self.file_collection.add(pipeline_file)
-        pipeline_file.file_update_callback = self._file_update_callback
+        if isinstance(pipeline_file, PipelineFile):
+            pipeline_file.file_update_callback = self._file_update_callback
```

Comment:
Just a thought... would it make sense to actually implement the "PipelineFile or local file path" behaviour in the `PipelineFile.__init__`? If an existing PipelineFile is passed, it would be copied (or returned as is), with any specified kwargs overriding the attributes already in there.
It would remove the need to deal with this case here, and for the Note you added in the doc string.
Apart from that, though, I can't think of any particular use case for it, so maybe not worth the effort.

Reply:
I had a look into this, but any other solution was getting a bit fiddly and complex, and the possibility of introducing bugs is high, so I'd say not worth the effort/risk in this case.

Reply:
:+1:  agreed.